api/post.go

```diff
@@ -576,186 +569,64 @@ func sendNotifications(c *Context, post *model.Post, team *model.Team, channel *
 				if post.UserId == userId && post.Props["from_webhook"] != "true" {
 					continue
 				}
-				sendEmail := true
-				if _, ok := profileMap[userId].NotifyProps["email"]; ok && profileMap[userId].NotifyProps["email"] == "false" {
-					sendEmail = false
-				}
-				if sendEmail && (profileMap[userId].IsAway() || profileMap[userId].IsOffline()) {
-					toEmailMap[userId] = true
-				} else {
-					toEmailMap[userId] = false
-				}
+
+				mentionedUserIds[userId] = true
 			}
 		}
 
-		for id := range toEmailMap {
+		for id := range mentionedUserIds {
 			go updateMentionCount(post.ChannelId, id)
 		}
 	}
 
-	if len(toEmailMap) != 0 {
-		mentionedUsers = make([]string, 0, len(toEmailMap))
-		for k := range toEmailMap {
-			mentionedUsers = append(mentionedUsers, k)
-		}
-
-		teamURL := c.GetSiteURL() + "/" + team.Name
+	mentionedUsersList := make([]string, 0, len(mentionedUserIds))
 
-		// Build and send the emails
-		tm := time.Unix(post.CreateAt/1000, 0)
+	senderName := profileMap[post.UserId].Username
 
-		for id, doSend := range toEmailMap {
+	for id := range mentionedUserIds {
+		mentionedUsersList = append(mentionedUsersList, id)
+	}
 
-			if !doSend {
-				continue
+	if utils.Cfg.EmailSettings.SendEmailNotifications {
+		for _, id := range mentionedUsersList {
+			userAllowsEmails := true
```

Comment:
Couldn't this just be `userAllowsEmails := profileMap[id].NotifyProps["email"] != "false"`? If they key doesn't exist in NotifyProps, it'll just return the empty string